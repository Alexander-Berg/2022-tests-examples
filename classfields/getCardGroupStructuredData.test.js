const _ = require('lodash');

const getCardGroupStructuredData = require('./getCardGroupStructuredData');
const listingMock = require('autoru-frontend/mockData/state/listing');
const cardGroupComplectations = require('autoru-frontend/mockData/state/cardGroupComplectations.mock.js');
const offer = require('auto-core/react/dataDomain/card/mocks/card.cars.mock');
const MockDate = require('mockdate');

describe('Разметка для групп карточек', () => {
    beforeEach(() => {
        MockDate.set('2021-12-01T21:00');
    });

    afterEach(() => {
        MockDate.reset();
    });

    it('Должен вернуть разметку', () => {
        const state = {
            ..._.cloneDeep(cardGroupComplectations),
            listing: {
                data: {
                    ...listingMock.data,
                    offers: [ {
                        ...offer,
                        additional_info: {
                            creation_date: '1553183449000',
                        },
                    } ],
                },
            },
            averageRating: {
                reviewsCount: 123,
                ratings: [
                    {
                        name: 'total',
                        value: 123.456,
                    },
                ],
            },
        };

        expect(getCardGroupStructuredData(state)).toMatchSnapshot();
    });

});
