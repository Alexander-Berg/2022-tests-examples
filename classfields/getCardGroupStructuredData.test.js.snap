// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`–†–∞–∑–º–µ—Ç–∫–∞ –¥–ª—è –≥—Ä—É–ø–ø –∫–∞—Ä—Ç–æ—á–µ–∫ –î–æ–ª–∂–µ–Ω –≤–µ—Ä–Ω—É—Ç—å —Ä–∞–∑–º–µ—Ç–∫—É 1`] = `
Array [
  Object {
    "@context": "https://schema.org",
    "@type": "https://schema.org/Product",
    "aggregateRating": Object {
      "@type": "AggregateRating",
      "bestRating": "5",
      "ratingValue": "123.5",
      "reviewCount": "123",
    },
    "brand": "FORD",
    "color": "–±–µ–ª—ã–π",
    "description": "–ê–≤—Ç–æ–º–æ–±–∏–ª—å –≤ –ò–î–ï–ê–õ–¨–ù–û–ú —Å–æ—Å—Ç–æ—è–Ω–∏–∏, –Ω–∞ 5+++++
–ù–∞ –ì–∞—Ä–∞–Ω—Ç–∏–∏!!!
–ó–∞–±–µ–≥–∞—è –≤–ø–µ—Ä–µ–¥ –ø—Ä–æ–¥–∞—é –∏–∑-–∑–∞ –Ω—É–∂–¥—ã –≤ –¥–µ–Ω—å‚Ä¶",
    "image": Object {
      "@context": "https://schema.org",
      "@type": "ImageObject",
      "author": "–î–º–∏—Ç—Ä–∏–π",
      "contentUrl": "http:picture-of-cat",
      "datePublished": "2019-03-21",
      "description": "Ford EcoSport I –±–µ–ª–æ–≥–æ —Ü–≤–µ—Ç–∞ –ø–æ —Ü–µ–Ω–µ 855¬†000¬†—Ä—É–±–ª–µ–π –≤ –ú–æ—Å–∫–≤–µ",
      "name": "Ford EcoSport I –±–µ–ª–æ–≥–æ —Ü–≤–µ—Ç–∞",
    },
    "name": "Ford EcoSport I",
    "offers": Object {
      "@context": "https://schema.org",
      "@type": "AggregateOffer",
      "highPrice": 45000,
      "lowPrice": 45000,
      "offerCount": 1,
      "priceCurrency": "RUR",
    },
    "productionDate": 2017,
  },
  Object {
    "@context": "https://schema.org",
    "@type": "http://schema.org/BreadcrumbList",
    "itemListElement": Array [
      Object {
        "@type": "ListItem",
        "item": Object {
          "@id": "https://autoru_frontend.base_domain/cars/toyota/new/?do_not_redirect=true",
          "@type": "Thing",
        },
        "name": "Toyota",
        "position": 1,
      },
      Object {
        "@type": "ListItem",
        "item": Object {
          "@id": "https://autoru_frontend.base_domain/cars/toyota/camry/new/?do_not_redirect=true",
          "@type": "Thing",
        },
        "name": "Camry",
        "position": 2,
      },
    ],
  },
  Object {
    "@context": "https://schema.org",
    "@type": "SaleEvent",
    "description": "–°—Ä–∞–≤–Ω–∏ —Å—Ç–æ–∏–º–æ—Å—Ç—å –û–°–ê–ì–û —É —Ä–∞–∑–Ω—ã—Ö –∫–æ–º–ø–∞–Ω–∏–π",
    "endDate": "2022-02-01T21:00:00+03:00",
    "location": Object {
      "@type": "Place",
      "name": "–ê–≤—Ç–æ.—Ä—É",
    },
    "name": "üîî –°—Ç—Ä–∞—Ö–æ–≤–∫–∞-–æ–Ω–ª–∞–π–Ω",
    "startDate": "2021-12-01T21:00:00+03:00",
  },
  Object {
    "@context": "https://schema.org",
    "@type": "SaleEvent",
    "description": "–ü—Ä–æ–≤–µ—Ä–∫–∞ –ü—Ä–æ–ê–≤—Ç–æ",
    "endDate": "2022-02-01T21:00:00+03:00",
    "location": Object {
      "@type": "Place",
      "name": "–ê–≤—Ç–æ.—Ä—É",
    },
    "name": "üîî –ü—Ä–æ–≤–µ—Ä–∫–∞ –ø–æ VIN",
    "startDate": "2021-12-01T21:00:00+03:00",
  },
  Object {
    "@context": "https://schema.org",
    "@type": "SaleEvent",
    "description": "–î–æ–±–∞–≤—å —Å–≤–æ–µ –∞–≤—Ç–æ –≤ –ì–∞—Ä–∞–∂ –∏ –ø–æ–ª—É—á–∞–π –±–æ–Ω—É—Å—ã",
    "endDate": "2022-02-01T21:00:00+03:00",
    "location": Object {
      "@type": "Place",
      "name": "–ê–≤—Ç–æ.—Ä—É",
    },
    "name": "üîî –°–∫–∏–¥–∫–∏ –≤ —Ç–≤–æ–µ–º –ì–∞—Ä–∞–∂–µ",
    "startDate": "2021-12-01T21:00:00+03:00",
  },
]
`;
